src/nvim/ui_bridge.c

```diff
@@ -88,6 +89,7 @@ UI *ui_bridge_attach(UI *ui, ui_main_fn ui_main, event_scheduler scheduler)
   uv_mutex_unlock(&rv->mutex);
 
   ui_attach_impl(&rv->bridge);
+  apply_autocmds(EVENT_UIATTACH, NULL, NULL, false, curbuf);
```

Comment:
v:event should be available here too.

Reply:
I assume this no longer applies since channel_id is only available in external attachment.

Reply:
Even so, `v:event` should always be available for a given event type, even if it's something like `v:event.channel_id = v:null`.

It would be weird if `v:event` were available for one UIAttach invocation but not another.

Reply:
`chanid = 0` would be the canonical sentinel value for "no channel" .

Reply:
@bfredl But we use `0` to mean "broadcast", so maybe it should be `-1` or `v:null`.

Reply:
Broadcast means "no specific channel". We also use 0 when connection/job failed, and thus no channel returned. Using different sentinels for "no channel" because the alternative is different is more confusing and surprising than helpful

Reply:
Ok.

Reply:
Pushing changes for this one. Any idea how to test this? I cant figure out how to catch this autocommand.